Description: meson 0.45 backward compatibility patch

   * meson 0.45 backward compatibility patch from Mario.

Author: Yuan-Chen Cheng <yc.cheng@canonical.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2019-08-01

--- fwupd-1.2.10.orig/meson.build
+++ fwupd-1.2.10/meson.build
@@ -1,7 +1,7 @@
 project('fwupd', 'c',
   version : '1.2.10',
   license : 'LGPL-2.1+',
-  meson_version : '>=0.47.0',
+  meson_version : '>=0.44.0',
   default_options : ['warning_level=2', 'c_std=c99'],
 )
 
@@ -115,8 +115,14 @@ test_link_args = [
   '-Wl,-z,now',
 ]
 foreach arg: test_link_args
-  if cc.has_link_argument(arg)
-    global_link_args += arg
+  if meson.version().version_compare('>=0.46.0')
+    if cc.has_link_argument(arg)
+      global_link_args += arg
+    endif
+  else
+    if cc.has_argument(arg)
+      global_link_args += arg
+    endif
   endif
 endforeach
 add_global_link_arguments(
